### DISCUSSION TITLE
Node-red Warnung

### DISCUSSION THREAD

**Discussion by totocotonio** [pixelit-project/PixelIt#349] [Help]
Hi,
seit dem neusten update nod-red habe ich folgende Warnung.

Node-RED:  1 Apr 15:02:38 - [warn] Context f1ca915489992660:3ae677fd.42e558  contains a circular reference that cannot be persisted
--

node-red.1 | 2024-04-01 15:02:38.750 | warn | Node-RED: 1 Apr 15:02:38 - [warn] Context b22d9ab5.1a84c8:3ae677fd.42e558 contains a circular reference that cannot be persisted

Diese Meldung kommt aus dem Flow der Matrix.

So wie ich das Interpretiere, ist das eine function,  die die Temperatur weiter gibt, denn wenn ich die funktion deaktiviere ist die Fehlermeldung weg.

Das hier steht in der function.

const data=Number(msg.payload).toFixed();
msg.payload=data+' °C';
return msg;

Das hat immer funktioniert.

Kann mir da einer helfen?

**Comment by foorschtbar** [comment]
Mach mal aus dem 'const' ein 'var'

**Comment by totocotonio** [comment]
Hi,
brachte keine Abhilfe!

**Comment by foorschtbar** [comment]
Dann kommt das aus von einer anderen stelle. Wenn du die Funktion deaktivierst, gibt sie auch an die nachfolgenden nodes keine daten mehr. Dort könnte auch der Fehler liegen.

**Comment by totocotonio** [comment]
ok
Aber warum erst nach einem Update. Da muss doch was passiert sein und ich bin doch bestimmt nicht der einzige.

Wie kann ich denn herausfinden, welcher Flow den Fehler verursacht?

**Comment by valbuz** [comment]
Ich habe dies auch seit dem Update.... :-(

**Comment by totocotonio** [comment]
Hm, wir können doch nicht die einzigen sein?

**Comment by foorschtbar** [comment]
Ich habe keine Problem feststellen können. Exportier makl bitte den Flow dann schau ich mir das bei gelegenheit mal an.

**Comment by totocotonio** [comment]
Hi, den kompletten Matrix Flow?

Kann man anhand der Nummer nicht irgendwie danach suchen?

Node-RED:  11 Apr 21:46:10 - [warn] Context f1ca915489992660:3ae677fd.42e558  contains a circular reference that cannot be persisted
--
node-red.1 | 2024-04-11 21:46:10.391 | warn | Node-RED:  11 Apr 21:46:10 - [warn] Context b22d9ab5.1a84c8:3ae677fd.42e558  contains a circular reference that cannot be persisted
